AndroidAnnotations/androidannotations-ormlite/ormlite-api/src/main/java/org/androidannotations/ormlite/api/OrmLiteInjectHelper.java

```diff
@@ -0,0 +1,8 @@
+package org.androidannotations.ormlite.api;
+
+/**
+ * @param <T> type to inject
+ */
+public interface OrmLiteInjectHelper<T> {
+	T inject();
+}
```

Comment:
What about the following: create a special inject helper which is a subclass of `InjectHelper`. In process(), it calls super.process first. Then if we should handle an exception, it removes the last statement from the block (you may have to use reflection for that), creates the try-catch and adds that black with try-bock containing the removed statement). Because the last statement int the block should be the method invocation or the assignment.